tests/aws/services/apigateway/test_apigateway_basic.py

```diff
@@ -1694,8 +1694,8 @@ def test_rest_api_multi_region(
 
     delete_rest_api(apigateway_client_eu, restApiId=api_eu_id)
     delete_rest_api(apigateway_client_us, restApiId=api_us_id)
-    testutil.delete_lambda_function(name=lambda_name, region_name="eu-west-1")
-    testutil.delete_lambda_function(name=lambda_name, region_name="us-west-1")
+    aws_client_factory(region_name="eu-west-1").lambda_.delete_function(FunctionName=lambda_name)
```

Comment:
Even better than manually deleting functions would be adopting the fixture `localstack.testing.pytest.fixtures.create_lambda_function`. It automatically cleans up dependencies (e.g., logs, role) properly. Further, it also waits until the function is active to simplify the test case.

Reply:
`localstack.testing.pytest.fixtures.create_lambda_function` doesn't offer the flexibility to override the AWS region :frowning_face: